caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -48,18 +48,14 @@ export function setFilterColumnOpts(filterColumnOpts) {
 }
 
 export function resetFormData() {
-  // Clear all form data when switching datasource
+  // Clear all form data when switching datasource or viz_type
   return { type: RESET_FORM_DATA };
```

Comment:
when a viz type changes, then the state will be updated which will trigger a rerender of the control panel sections. https://github.com/airbnb/caravel/blob/master/caravel/assets/javascripts/explorev2/components/ControlPanelsContainer.jsx#L29

the sections and fields that get rendered are dynamic depending on the viz type. 

so what i am getting at is that the field defaults should be stored in the field definitions, and we shouldn't need to reset the whole form at any point, react will take care of triggering the rerender and will render the appropriate fields/default when a new viz type is updated.